src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/DbReference.scala

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsde.workbench.leonardo.db
+
+import java.sql.SQLTimeoutException
+
+import akka.actor.ActorSystem
+import com.google.common.base.Throwables
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import liquibase.{Contexts, Liquibase}
+import liquibase.database.jvm.JdbcConnection
+import liquibase.resource.{ClassLoaderResourceAccessor, ResourceAccessor}
+import org.broadinstitute.dsde.workbench.leonardo.config.LiquibaseConfig
+import slick.basic.DatabaseConfig
+import slick.dbio.DBIO
+import slick.jdbc.{JdbcBackend, JdbcDataSource, JdbcProfile, TransactionIsolation}
+import net.ceedubs.ficus.Ficus._
+import sun.security.provider.certpath.SunCertPathBuilderException
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object DbReference extends LazyLogging {
+
+  private def initWithLiquibase(dataSource: JdbcDataSource, liquibaseConfig: LiquibaseConfig, changelogParameters: Map[String, AnyRef] = Map.empty): Unit = {
+    val dbConnection = dataSource.createConnection()
+    try {
+      val liquibaseConnection = new JdbcConnection(dbConnection)
+      val resourceAccessor: ResourceAccessor = new ClassLoaderResourceAccessor()
+      val liquibase = new Liquibase(liquibaseConfig.changelog, resourceAccessor, liquibaseConnection)
+
+      changelogParameters.foreach { case (key, value) => liquibase.setChangeLogParameter(key, value) }
+      liquibase.update(new Contexts())
+    } catch {
+      case e: SQLTimeoutException =>
+        val isCertProblem = Throwables.getRootCause(e).isInstanceOf[SunCertPathBuilderException]
+        if (isCertProblem) {
+          val k = "javax.net.ssl.keyStore"
+          if (System.getProperty(k) == null) {
+            logger.warn("************")
+            logger.warn(s"The system property '${k}' is null. This is likely the cause of the database connection failure.")
+            logger.warn("************")
+          }
+        }
+        throw e
+    } finally {
+      dbConnection.close()
+    }
+  }
+
+  def init(config: Config, system: ActorSystem)(implicit executionContext: ExecutionContext): DbReference = {
+    val db = "mysql"
+    val dbConfig = DatabaseConfig.forConfig[JdbcProfile](db, config)
+
+    system.registerOnTermination {
+      dbConfig.db.shutdown
+    }
+
+    val liquibaseConf = config.as[LiquibaseConfig]("liquibase")
+    if (liquibaseConf.initWithLiquibase)
+      initWithLiquibase(dbConfig.db.source, liquibaseConf)
+
+    DbReference(dbConfig)
+  }
+}
+
+case class DbReference(private val dbConfig: DatabaseConfig[JdbcProfile])(implicit val executionContext: ExecutionContext) extends AllComponents {
```

Comment:
I did not include that because it appeared to be a workaround to a bug in an older version of Slick.  I can add it if you think it's worthwhile independently of that.